client/python/lib/test/test_tensors.py

```diff
@@ -77,154 +77,270 @@ def test_make_ndarray_invalid_type():
-    assert tensor_proto.tensor_shape == TensorShapeProto(dim = [TensorShapeProto.Dim(size=1)])
+    if params.get("dtype"):
+        assert tensor_proto.dtype == params.get("dtype")
+    else:
+        assert tensor_proto.dtype == NP_TO_TENSOR_MAP.get(np_values.dtype.type).TensorDtype
+
+    if params.get("shape"):
+        assert tensor_proto.tensor_shape == TensorShapeProto(dim = [TensorShapeProto.Dim(size=x) for x in params["shape"]])
+    else:
+        assert tensor_proto.tensor_shape == TensorShapeProto(dim = [TensorShapeProto.Dim(size=3)])
+
+@pytest.mark.parametrize("params", [
+    {"values" : 5.0},
+    {"values" : 5.0, "shape" : [1]},
+    {"values" : 5.0, "dtype" : DataType.DT_INT32},
+    {"values" : 5.0, "dtype" : DataType.DT_FLOAT},
+    {"values" : 5.0, "shape" : [1], "dtype" : DataType.DT_INT32},
+    {"values" : 5.0, "shape" : [1], "dtype" : DataType.DT_FLOAT},
+    {"values" : np.float32(5.0)},
+    {"values" : np.float32(5.0), "shape" : [1]},
+    {"values" : np.float32(5.0), "dtype" : DataType.DT_INT32},
+    {"values" : np.float32(5.0), "dtype" : DataType.DT_FLOAT},
+    {"values" : np.float32(5.0), "shape" : [1], "dtype" : DataType.DT_INT32},
+    {"values" : np.float32(5.0), "shape" : [1], "dtype" : DataType.DT_FLOAT},
+    {"values" : np.float32(5.0), "shape" : [1], "dtype" : DataType.DT_DOUBLE},
+])
+def test_make_tensor_proto_valid_scalar(params):
+    tensor_proto = make_tensor_proto(**params)
+    np_dtype = TENSOR_TO_NP_MAP.get(params.get("dtype"))
+    np_values = np.array(params["values"], dtype=np_dtype)
+    
+    if NP_TO_TENSOR_MAP.get(np_values.dtype.type).TensorDtype == DataType.DT_INT32:
+        assert tensor_proto.int_val == np_values
+    elif NP_TO_TENSOR_MAP.get(np_values.dtype.type).TensorDtype == DataType.DT_FLOAT:
+        assert tensor_proto.float_val == np_values
+    else:
+        assert tensor_proto.double_val == np_values
+    
+    if params.get("dtype"):
+        assert tensor_proto.dtype == params.get("dtype")
+    else:
+        assert tensor_proto.dtype == NP_TO_TENSOR_MAP.get(np_values.dtype.type).TensorDtype
+    
+    if params.get("shape"):
+        assert tensor_proto.tensor_shape == TensorShapeProto(dim = [TensorShapeProto.Dim(size=x) for x in params["shape"]])
+    else:
+        assert tensor_proto.tensor_shape == TensorShapeProto(dim = [TensorShapeProto.Dim(size=1)])
+
+@pytest.mark.parametrize("params", [
+    {"values" : bytes([0x13, 0x00, 0x00, 0x00, 0x08])},
+    {"values" : bytes([0x13, 0x00, 0x00, 0x00, 0x08]), "shape" : [1]},
+    {"values" : bytes([0x13, 0x00, 0x00, 0x00, 0x08]), "dtype" : DataType.DT_STRING},
+    {"values" : bytes([0x13, 0x00, 0x00, 0x00, 0x08]), "dtype" : DataType.DT_INT8},
+    {"values" : bytes([0x13, 0x00, 0x00, 0x00, 0x08]), "shape" : [5], "dtype" : DataType.DT_INT8},
+    {"values" : bytes([0x13, 0x00, 0x00, 0x00, 0x08]), "shape" : [1], "dtype" : DataType.DT_STRING},
+])
+def test_make_tensor_proto_valid_string(params):
```

Comment:
We get a lot of if..else... blocks here. Is it possible to remove them? For example by creating a mapping (condition -> expected_value) instead of adding additional comparisons.